### PR TITLE
Refactoring UberstreamLog Cleanup job to use SQL and adding processingDate as partition column to streaming_log tables

### PR DESCRIPTION
**Pull request by gcoffmanQ**

### CODE REVIEW

File: UberStream_Log_Cleanup.py [hunk 1]
```diff
@@ -11,21 +11,43 @@
 CLEANUP_TTL_VAL = dbutils.widgets.get(CLEANUP_TTL_KEY)
 STREAMING_LOG_TABLES = ["action_stream", "transition_stream", "traveler_stream"]
 
-MAX_OFFSET_PER_PARTITION_QUERY = """
-SELECT group_by_data.topic, group_by_data.partition, group_by_data.offset, outer_data.timestamp, outer_data.data, outer_data.batchid, outer_data.processingTimestamp, outer_data.processingDate
+DELETE_QUERY = """
+DELETE FROM streaming_logs.[STREAM_TABLE] WHERE
+processingTimestamp < '[CUTOFF_TIMESTAMP]';
```
Comment (cfluture) [inline UberStream_Log_Cleanup.py:16]: have to use processingDate in order to take advantage of partitioning, precision is not important as we are cleaning logs

File: Über Stream.scala [hunk 2]
```diff
@@ -171,22 +171,22 @@ def addColumnToTable(orgId:Long, table:String, column:String, columnType:String)
   """)
 }
 
-def addDateColumnToTable(table: String, columnNameToCheck: String) : Unit = {
+def addDateColumnToTable(table: String, dateCol: String) : Unit = {
   val df = spark.sql(f"SELECT * FROM streaming_logs.$table%s")
```
Comment (cfluture) [inline Über Stream.scala:175]: potentially expensive, use DESCRIBE TABLE instead, this will only call the meta store no matter how big the table is

Comment (gcoffmanQ) [reply Über Stream.scala]: Databricks SQL didn't have DESCRIBE TABLE so I used their function SHOW COLUMNS and then performed a where on that to find the dateColumn

File: Über Stream.scala [hunk 3]
```diff
@@ -171,29 +171,34 @@ def addColumnToTable(orgId:Long, table:String, column:String, columnType:String)
...
-    println(f"$columnNameToCheck%s column found in table streaming_logs.$table%s")
+def addDateColumnToTable(table: String, dateCol: String) : Unit = {
+  val dateColPresent = spark.sql(f"SHOW COLUMNS IN streaming_logs.$table%s").where(f"col_name = '$dateCol%s'")
+  if (dateColPresent.count() > 0) {
+    println(f"$dateCol%s column found in table streaming_logs.$table%s")
     return
   } else {
-    println(f"$columnNameToCheck%s column not found in table streaming_logs.$table%s. Saving current table to streaming_logs.$table%s_backup before adding $columnNameToCheck%s")
+    println(f"$dateCol%s column not found in table streaming_logs.$table%s. Saving current table to streaming_logs.$table%s_backup before adding $dateCol%s")
+    val df = spark.sql(f"SELECT * FROM streaming_logs.$table%s")
```
Comment (cfluture) [inline Über Stream.scala:181]: there is no need to read as a dataframe, use CREATE TABLE AS SELECT query instead

File: Über Stream.scala [hunk 4]
```diff
@@ -171,29 +171,34 @@ def addColumnToTable(orgId:Long, table:String, column:String, columnType:String)
...
     df
       .write
       .mode("overwrite")
       .saveAsTable(f"streaming_logs.$table%s_backup")
       
     df
-      .withColumn(columnNameToCheck, to_date(col("processingTimestamp"),"yyyy-MM-dd"))
```
Comment (cfluture) [inline Über Stream.scala:187]: use CREATE TABLE AS SELECT query instead

**Review by cfluture — CHANGES_REQUESTED** [review]

**Review by cfluture — APPROVED** [review]

**Pull request closed without merging**